src/_igraph/graphobject.c

```diff
@@ -13748,6 +13748,149 @@ PyObject *igraphmodule_Graph_community_leiden(igraphmodule_GraphObject *self,
   return error ? NULL : Py_BuildValue("Nd", res, (double) quality);
 }
 
+/**
+ * Voronoi clustering
+ */
+PyObject *igraphmodule_Graph_community_voronoi(igraphmodule_GraphObject *self,
+                                               PyObject *args, PyObject *kwds) {
+    static char *kwlist[] = {"modularity", "lengths", "weights", "mode", "radius", NULL};
+    PyObject *lengths_o = Py_None, *weights_o = Py_None;
+    PyObject *mode_o = Py_None;
+    PyObject *radius_o = Py_None;
+    PyObject *modularity_o = Py_None;
+    igraph_vector_t *lengths_v = NULL;
+    igraph_vector_t *weights_v = NULL;
+    igraph_vector_int_t membership_v, generators_v;
+    igraph_neimode_t mode = IGRAPH_ALL;
+    igraph_real_t radius = -1.0;  /* negative means auto-optimize */
+    igraph_real_t modularity = IGRAPH_NAN;
+    PyObject *membership_o, *generators_o, *result_o;
+    igraph_bool_t return_modularity = false;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|OOOOO", kwlist,
+                                     &modularity_o, &lengths_o, &weights_o, &mode_o, &radius_o))
+        return NULL;
+
+    if (modularity_o != Py_None){
+      modularity = (igraph_real_t)PyFloat_AsDouble(modularity_o);
+    }
+    else {
+      return_modularity = true;
+    }
+
+    /* Handle mode parameter */
+    if (mode_o != Py_None) {
+        if (igraphmodule_PyObject_to_neimode_t(mode_o, &mode))
+            return NULL;
+    }
+
+    /* Handle radius parameter */
+    if (radius_o != Py_None) {
+        if (PyFloat_Check(radius_o)) {
+            radius = PyFloat_AsDouble(radius_o);
+        } else if (PyLong_Check(radius_o)) {
+            radius = PyLong_AsDouble(radius_o);
+        } else {
+            PyErr_SetString(PyExc_TypeError, "radius must be a number or None");
+            return NULL;
+        }
+        if (PyErr_Occurred()) return NULL;
+    }
+
+    /* Handle lengths parameter */
+    if (lengths_o != Py_None) {
```

Comment:
No need for `Py_None` checks here either -- `igraphmodule_attrib_to_vector_t` handles `Py_None` and sets the output pointer (i.e.` lengths_v` here) to `NULL` so you know that no weights were supplied.